### DISCUSSION TITLE
xterm js with react

### DISCUSSION THREAD

**Discussion by sagoresarker** [sourcelair/xterm.js#4895] [Q&A] [answered]
Hello, this is the first time I'm working with xterm.js in React. I tried to install 'xterm-for-react' from the npm library, but it threw so many errors.
Would you like to share any suggestions that will help me configure xterm with React?

**Comment by tisilent** [comment]
Instantiating and destroying xtermjs in useEffect.

**Comment by x1unix** [comment, marked as answer]
@sagoresarker that library is quite abandoned and uses old React and xterm packages.

I suggest just take XTerm component from [here](https://github.com/jaxtonw/xterm-for-react18/blob/master/src/XTerm.tsx) and copy to your project.

It supports React 18 and should work without any problems.

**Reply by sagoresarker** [reply]
Thanks
